src/promptflow-evals/promptflow/evals/evaluate/_eval_run.py

```diff
@@ -306,6 +309,25 @@ def _log_warning(self, failed_op: str, response: requests.Response) -> None:
             f"{response.text=}."
         )
 
+    def _check_state_and_log(self, action : str) -> bool:

```

Comment:
Should we fail the call instead of just logging a warning if the state is incorrect? 

I feel raising an exception here is better. And, customers will see the final error message. To avoid confusion, let's use a common message like "Unable to {action} due to Run status={status}" instead of saying "please do something...".

Reply:
Fixed the message.
The run is not doing any further actions if incorrect state is reached and logs the corresponding error message according to previous suggestions. Should we still raise the error?

Reply:
When user run into incorrect state, it must be bug that developer should fix. Thus, I think raising the error is better than logging a warning message.